extensions/scarb-doc/src/docs_generation/markdown/traits.rs

```diff
@@ -153,6 +169,26 @@ impl MarkdownDocItem for Trait {
     }
 }
 
+pub fn mark_items_if_duplicated_name<'a, T: TopLevelMarkdownDocItem + 'a>(
+    items: &'a [&'a T],
+) -> Vec<(&&'a T, bool)> {
+    let mut names_counter = HashMap::<String, u32>::new();
+    for item in items {
+        *names_counter.entry(item.name().to_string()).or_insert(0) += 1;
+    }
+
+    items
+        .iter()
+        .map(|item| {
+            let is_duplicated = match names_counter.get(item.name()) {
+                Some(value) => *value > 1,
```

Comment:
Why not just use `HashSet` instead of a map?

Reply:
We need to know whether there are more than 1 item with the same name. With Set, we are unable to tell if there's a single item (not duplicated), or more of them. Or I missed something? We would need 2 sets, the `observed_set` and `duplicated_set`

Reply:
Hmm, I am not sure what I envisioned. HashMap is good :D